### PR TITLE
Updated text generation to return gen_text object

### PR DESCRIPTION
**Pull request by johntmyers**
This is change in the object that gets returned from `generate_text()`. Instead of returning a `dict` we will return the underlying `gen_text` object. It has the same data as before but are now accessed as attrs instead of dictionary keys.

Example:

```python
for line in generate_text(config, line_validator=validate_record):
    print(line.text). # instead of line["text"]
```

This is to keep the text object more Pythonic for when this package is used by the premium Gretel service for data synthesis. We can still grab a dictionary by doing`line.as_dict()`

This change does not impact the example notebook.

Also added in `black` formatting to the other files for a little cleanup.

### CODE REVIEW

**Comment by johntmyers** [comment]
Quick update--having reviewed #14 this would be good future proofing for these types of features too as the `gen_text` object can carry the knowledge of the delimination char and provide the necessary methods to break the record into a list, etc. I may make these updates now so we have the interface available for the future.

File: src/gretel_synthetics/generate.py [hunk 1]
```diff
@@ -175,18 +198,17 @@ def _predict_chars(model: tf.keras.Sequential,
...
+        predicted_id = tf.random.categorical(predictions, num_samples=1)[-1, 0].numpy()
 
         # We pass the predicted word as the next input to the model
         # along with the previous hidden state
         input_eval = tf.expand_dims([predicted_id], 0)
         sentence_ids.append(int(predicted_id))
 
         decoded = sp.DecodeIds(sentence_ids)
-        decoded = decoded.replace('<c>', ',')
+        decoded = decoded.replace("<c>", ",")
```
Comment (zredlined) [inline src/gretel_synthetics/generate.py:209]: @johntmyers we will want whatever delimiter exists in the input data to be encoded as a special token here. Two options would be: a) creating a dictionary mapping of common delimiters and tags that are passed to the sentencepiecetokenizer, and then reversed later. Or b) make the delimiter a BaseConfig item and creating a user defined token from that. See: https://github.com/gretelai/gretel-synthetics/issues/14

Comment (johntmyers) [reply src/gretel_synthetics/generate.py:209]: The delimeter is a config option, see the `field_delimeter` attr that is defined on the base dataclass. Why do we need a different token? Assuming there's only one delimeter provided, can we just use the same token regardless of the delimeter char provided?

Comment (johntmyers) [reply src/gretel_synthetics/generate.py:209]: I'm also assuming we need to update the annotation function that does this:

```python
  with open(store.input_data_path, 'r', encoding='utf-8', errors='replace') as infile:
        for line in infile:
            if store.max_lines and len(training_text) >= store.max_lines:
                break
            line = line.strip().replace(",", "<c>")
            training_text.append(line)
```

Comment (zredlined) [reply src/gretel_synthetics/generate.py:209]: My bad- I did not see the `field_delimiter` option in the base dataclass. 

As described above we'll need to replace the replace the delimiter once as a special character on annotation, specify the corresponding user defined token to the tokenizer, and then reverse the token with the delimiter in the text generation functions

**Pull request merged**